src/test/java/com/microsoft/sqlserver/jdbc/connection/ConnectionTest.java

```diff
@@ -43,4 +43,32 @@ public void testConnections() throws SQLException {
         try (Connection con = ds.getConnection()) {
         }
     }
+    
+    @Test
+    public void testConnectWithIPAddressPreference () throws SQLException {
+        SQLServerDataSource ds = new SQLServerDataSource();
+        ds.setURL(connectionString);
+        ds.setIPAddressPreference("IPv4First");
+        
+        try (Connection con = ds.getConnection()) {
+        }
+        
+        ds.setIPAddressPreference("IPv6First");
+        
+        try (Connection con = ds.getConnection()) {
+        }
+        
+        ds.setIPAddressPreference("UsePlatformDefault");
+        
+        try (Connection con = ds.getConnection()) {
+        }
+        
+        ds.setIPAddressPreference("Bogus");
+        
+        try (Connection con = ds.getConnection()) {
+        }
+        catch (Exception e){
+            assert e.getMessage().equals("Invalid option for connection string option IPAddressPreference, the available options are: IPv4First, IPv6First and UsePlatformDefault");

```

Comment:
we should be comparing against the string in the driver, use TestResource.getResource. Check other tests to see how this is done

Reply:
Agreed, changes made.